src/coreclr/tools/aot/ILCompiler.ReadyToRun/CodeGen/ReadyToRunObjectWriter.cs

```diff
@@ -307,6 +317,37 @@ public void EmitPortableExecutable()
                     }
                 }
 
+                // Emit cold method nodes to end of execution section.
+                foreach (ObjectNode node in methodColdCodeNodes)
+                {
+                    ++nodeIndex;
+
+                    if (node == null)
+                    {
+                        continue;
+                    }
+
+                    ObjectData nodeContents = node.GetData(_nodeFactory);
+                    string name = null;
+
+                    if (_mapFileBuilder != null)
```

Comment:
Could we move this to a helper method to avoid duplicating the generic hackery?

Reply:
Sure thing, just pushed a revision. Thank you for the review!